tests/components/qwikswitch/test_init.py

```diff
@@ -106,3 +114,232 @@ async def test_sensor_device(hass, aioclient_mock):
     assert state_obj.state == "416"
 
     listen_mock.stop()
+
+
+async def test_switch_device(hass, aioclient_mock, qs_devices):
+    """Test a switch device."""
+
+    async def get_devices_json(method, url, data):
+        return AiohttpClientMockResponse(method=method, url=url, json=qs_devices)
+
+    config = {"qwikswitch": {"switches": ["@a00001"]}}
+    aioclient_mock.get("http://127.0.0.1:2020/&device", side_effect=get_devices_json)
+    listen_mock = MockLongPollSideEffect()
+    aioclient_mock.get("http://127.0.0.1:2020/&listen", side_effect=listen_mock)
+    assert await async_setup_component(hass, QWIKSWITCH, config)
+    hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+    await hass.async_block_till_done()
+
+    state_obj = hass.states.get("switch.switch_1")
+    assert state_obj.state == "off"
+
+    aioclient_mock.mock_calls.clear()
+    aioclient_mock.get("http://127.0.0.1:2020/@a00001=100")
+    await hass.services.async_call(
+        "switch", "turn_on", {"entity_id": "switch.switch_1"}, blocking=True
+    )
+    await asyncio.sleep(0.01)
+    assert (
+        "GET",
+        URL("http://127.0.0.1:2020/@a00001=100"),
+        None,
+        None,
+    ) in aioclient_mock.mock_calls
+
+    qs_devices[0]["val"] = "ON"
+    state_obj = hass.states.get("switch.switch_1")
+    assert state_obj.state == "off"
+    listen_mock.queue_response(json={"id": "@a00001", "cmd": ""},)
+    await hass.async_block_till_done()
+    state_obj = hass.states.get("switch.switch_1")
+    assert state_obj.state == "on"
+
+    aioclient_mock.mock_calls.clear()
+    aioclient_mock.get("http://127.0.0.1:2020/@a00001=0")
```

Comment:
not sure i understand. what dict are you referring to? This is just to see that the library calls "http://127.0.0.1:2020/@a00001=0"

Reply:
See line 136 above. The library calls it correctly, but the result is a failure
Now here it does not matter since you dont test the hass sensor state, but above you can see that the sensor never updates

Reply:
i know. the goal in this isn't to test the library itself. I am basically running four scenarios:
1. switch is off, library turns it on, check that state is updated in hass
2. switch is off, hass asks to turn device on, check that library issues the http command to turn on
3/4. same thing when the switch is on.
IMO, the goal in the unit tests isn't to simulate the full environment, but to check that small steps cause the right expected actions. If it is a simple return code, we can add it easily, but prefer not to add something that will complicate the test and make it less readable / maintainable. Let me know if you want to add something and what the attributes of the dict should be

Reply:
All set commands should return a dict with a `data` key. If you don’t the library creates tasks in the background to retry... 

```python
aioclient_mock.get("http://127.0.0.1:2020/@a00001=on/off", {"id":"@a00001","data":"anything"})
```

Please don’t try to work against the library. If you want to test different scenarios, do it in different test functions.

Reply:
that makes sense. I believe that i fixed it. can you please take a look at the latest?